app/views/home.py

```diff
@@ -72,13 +74,49 @@ def send_email():
 
 @bp.route("/appointment")
 def appointment():
-    booked_appointments = appintments_model.get_booked_slots()
+    booked_appointments = appointment_model.get_booked_slots()
     timeslots = [a["timeslot"].date() for a in booked_appointments]
-    print(timeslots)
-    data = {"appointments": timeslots}
+    locked_slots = {}
+    # locked_slots: Is for dynamic lock days on the calendar in the front-end
+    for a in booked_appointments:
+        service = services_model.get(a["service_id"])
+        if service["max_slot"]:
+            if service["id"] not in locked_slots:
+                locked_slots[service["id"]] = []
+            locked_slots[service["id"]].append(
+                str(a["timeslot"].date())
+            )
+    shops = shops_model.all() or []
+    services = services_model.all() or []
+    data = {
+        "shops": shops,
+        "services": services,
+        "appointments": timeslots,
+        "locked_slots": locked_slots
+    }
     return render_template("appointment.html", config=config, data=data)
 
 
+@bp.route("/appointment", methods=["POST"])
+def save_appointment():
+    appointment_model.insert(
+        shop_id=None,
+        service_id=request.form.get("service"),
+        timeslot=datetime.strptime(request.form.get("timeslot"), "%Y-%m-%d"),
+        vehicle=request.form.get("vehicle"),
+        name=request.form.get("name"),
+        email=request.form.get("email"),
+        description=request.form.get("description")
+    )
+    sendgrid.send(
+        config["TO_EMAIL_ADDRESS"],
+        config["TO_EMAIL_ADDRESS"],
+        "New appointment has been created",
+        "New appointment has been created"
```

Comment:
I think the body should be something different than just a "new appointment has been created" we should add more details like the service name, and all other information